atest/testdata/standard_libraries/builtin/wait_until_keyword_succeeds.robot

```diff
@@ -115,3 +125,13 @@ Wait Until Inside User Keyword
 Timeouted UK with Wait Until KW
     [Timeout]    ${timeout}
     Wait Until Keyword Succeeds    100ms    10ms    Fail    Error in timeouted UK
+
+Get Length On Returned Variable Works On Last Attempt
+    ${myvar} =    return_value_after_retries    Hello
+    Get Length    ${myvar}
+    Fail    Test PASSES : variable is no more None
```

Comment:
OK Thx. I'll have a look then ;)
Notice that, at the moment, using the proposed keyword in WUKS :
```robot
*** Keywords ***
Example
    Log    ${dynamically set variable}    # doesn't exist when this keyword is executed first time
    Set Test Variable    ${dynamically set variable}    value
```
the line `Set Test Variable    ${dynamically set variable}    value` is never executed.

Still analysing this